src/icmp/Icmp4.cc

```diff
@@ -115,9 +116,10 @@ Icmp4::SendEcho(Ip::Address &to, int opcode, const char *payload, int len)
     // Construct ICMP packet data content
     echo = reinterpret_cast<icmpEchoData *>(reinterpret_cast<char *>(pkt) + sizeof(*icmp));
     echo->opcode = (unsigned char) opcode;
-    memcpy(&echo->tv, &current_time, sizeof(struct timeval));
+    const auto now = std::chrono::system_clock::now();
+    memcpy(&echo->tv, &now, sizeof(now));
```

Comment:
Are you sure? According to [cppreference](https://en.cppreference.com/w/cpp/chrono/system_clock/now), now() returns a `std::chrono::time_point`, I don't think it can be safely copied to a struct timeval like this

Reply:
Oops, forgot to change `tv` type. Done now in 2dc4b66ed9